tests/metrics/map_mem.sh

```diff
@@ -0,0 +1,287 @@
+	for cmd in ${req_tools[*]};do
+		command -v "$cmd" &> /dev/null
+		res=$?
+		if [ $res -eq 0 ];then
+			echo "command: $cmd: yes"
+			continue
+		fi
+
+		echo  "error: command $cmd not available"
+	done
+
+	# clean env
+	echo "cleaning env"
+	rm -rf *.log
+	rm -rf *.json
+}
+
+# dump memory avg to JSON file
+function dump_avg_JSON()
+{
+	pss=$1
+	shared=$2
+	private=$3
+	count=$4
+
+	pss_avg=$(echo "scale=2; $pss / $count" | bc)
+	shared_avg=$(echo "scale=2; $shared / $count" | bc -l )
+	private_avg=$(echo "scale=2; $private / $count" | bc -l)
+
+	echo -e "\t\t\"pss\": $pss_avg," >> $JSON_file
+	echo -e "\t\t\"shared\": $shared_avg," >> $JSON_file
+	echo -e "\t\t\"private\": $private_avg" >> $JSON_file
+}
+
+# it creates JSON file
+function create_json()
+{
+	local mapping=""
+	local pss_mem=0
+	local shared_mem=0
+	local private_mem=0
+	local count=0
+
+	# accumulators
+	local shared_sum=0
+	local private_sum=0
+	local pssmem_sum=0
+
+	declare -A memory=(
+		["mapping"]=1
+		["pssmem"]=2
+		["shared"]=3
+		["private"]=4)
+
+	JSON_file="${PROC}.json"
+	output="$(cat "$SAMPLES_FILE" | sort)"
+
+	# start JSON file creation
+	echo "{" > "$JSON_file"
+
+	while read line;do
+		mapping=$(echo "$line" | awk -F "," -v column=${memory["mapping"]} '{print $column}')
+		pss_mem=$(echo "$line" | awk -F "," -v column=${memory["pssmem"]} '{print $column}')
+		shared_mem=$(echo "$line" | awk -F "," -v column=${memory["shared"]} '{print $column}')
+		private_mem=$(echo "$line" | awk -F "," -v column=${memory["private"]} '{print $column}')
+		count=$(( $count + 1 ))
+
+		# last state
+		if [ -z "$last_mapping" ];then
+			last_mapping="$mapping"
+		fi
+
+		# dump data
+		if [ "$mapping" != "$last_mapping" ];then
+			echo -e "\t\"$last_mapping\": {" >> $JSON_file
+			dump_avg_JSON $pssmem_sum $shared_sum $private_sum $count
+			echo -e "\t}," >> $JSON_file
+			shared_sum=0
+			private_sum=0
+			pssmem_sum=0
+			count=0
+			last_mapping="$mapping"
+		fi
+
+		# accumulate data
+		shared_sum=$(( $shared_sum + $shared_mem ))
+		private_sum=$(( $private_sum + $private_mem ))
+		pssmem_sum=$(( $pssmem_sum + $pss_mem ))
+
+	done <<< $(echo -e "$output")
+
+	# dump buffers
+	echo -e "\t\"$last_mapping\": {" >> $JSON_file
+	dump_avg_JSON $pssmem_sum $shared_sum $private_sum $count
+	echo -e "\t}" >> $JSON_file
+
+	# Finish JSON file creation
+	echo "}" >> $JSON_file
+}
+
+# take one sample of mapping
+function take_map_sample()
+{
+	local ps_name=$1
+	local format_columns=23
+	local last_mapping=""
+	local mapping=""
+
+	# memory values
+	local pssmem=0
+	local pssmem_sum=0
+	local shared_memory=0
+	local shared_clean=0
+	local shared_dirty=0
+	local shared_hugep=0
+	local private_clean=0
+	local private_dirty=0
+	local private_hugep=0
+	local private_memory=0
+
+	# accumulators
+	local shared_sum=0
+	local private_sum=0
+	local anon_pssmem=0
+	local anon_shared=0
+	local anon_private=0
+
+	# types of memory
+	declare -A memory=(
+		["pssmem"]=8
```

Comment:
Again, an explanatory comment on what these numbers are would be useful from a maintenance perspective.